QA Report: XNU x64 CMake build.md Failure

## QA Reproduction
Tue Aug 18 12:46:48 EDT 2026

## Host
Darwin Kernel Version 24.6.0: Tue Jul 21 20:50:07 PDT 2026; root:xnu-11417.140.69.711.44~1/RELEASE_X86_64 x86_64

### Steps
Follow the build.md instructions for macOS on x64 Intel Host

### User Experience

```
-- ############################################################
-- ## Developer tools for ICC color profiles ##
-- ############################################################
-- ## Configuration and Build started at 2026-08-18 11:17:18 ##
--
-- The C compiler identification is AppleClang 17.0.0.17000604
-- The CXX compiler identification is AppleClang 17.0.0.17000604
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Error at CMakeLists.txt:249 (message):
  Refusing to use the repository root as the CMake build directory.  Use -B
  out/<name>, run legacy builds from Build/, or use a preset with an isolated
  binaryDir.


-- Configuring incomplete, errors occurred!
Command line invocation:
    /Applications/Xcode.app/Contents/Developer/usr/bin/xcodebuild -project RefIccMAX.xcodeproj
```

--- a/Build/Cmake/CMakePresets.json
+++ b/Build/Cmake/CMakePresets.json
@@ -55,6 +55,18 @@
       "displayName": "Windows ClangCL (VS2022) QA flags"
     },
     {
+      "name": "vs2026-x64",
+      "inherits": "vs2022-x64",
+      "displayName": "Windows MSVC (VS2026)",
+      "description": "Builds with the Visual Studio 2026 generator and v145 toolset for machines that have the VS2026 build tools but not the VS2022 v143 component.",
+      "generator": "Visual Studio 18 2026"
+    },
+    {
+      "name": "vs2026-x64-qa-flags",
+      "inherits": ["vs2026-x64", "qa-flags-base"],
+      "displayName": "Windows MSVC (VS2026) QA flags"
+    },
+    {
       "name": "mingw-ucrt64-base",
       "inherits": "default-base",
       "hidden": true,

--- a/docs/build.md
+++ b/docs/build.md
@@ -1,6 +1,6 @@
 # Building iccDEV
 
-iccDEV requires C++17, CMake 3.18 or newer, and the image/XML/JSON dependencies
+iccDEV requires C++17, CMake 3.23 or newer, and the image/XML/JSON dependencies
 listed below. Maintainer-level sanitizer, Docker, and CMake policy details live in
 `.github/instructions/build-system.instructions.md`.
 
@@ -123,11 +123,14 @@ layout as the MSVC preset.
 ## Windows MSVC 2026
 
 The `vs2026-x64` preset targets the Visual Studio 18 2026 generator and v145
-toolset while preserving the VS 2022 preset's vcpkg and runtime layout:
+toolset while preserving the VS 2022 preset's vcpkg and runtime layout. CMake
+3.23 or newer is required for the preset schema. Set `VCPKG_ROOT` to the
+Visual Studio 2026 vcpkg installation before configuring:
 
 ```cmd
 git clone https://github.com/InternationalColorConsortium/iccDEV.git iccdev
 cd iccdev
+set "VCPKG_ROOT=C:\Program Files\Microsoft Visual Studio\2026\Community\VC\vcpkg"
 cmake --preset vs2026-x64 -S Build/Cmake -B out/vs2026-x64
 cmake --build out/vs2026-x64 --config Release -- /m /maxcpucount
 ```

--- a/docs/build.md
+++ b/docs/build.md
@@ -18,7 +18,7 @@ listed below. Maintainer-level sanitizer, Docker, and CMake policy details live 
 |----------|----------|
 | Ubuntu | `libpng-dev libjpeg-dev libtiff-dev libxml2-dev zlib1g-dev libwxgtk3.2-dev libwxgtk-media3.2-dev libwxgtk-webview3.2-dev wx-common wx3.2-headers nlohmann-json3-dev cmake make ninja-build` |
 | macOS | `libpng jpeg-turbo libtiff libxml2 zlib wxwidgets nlohmann-json` |
-| Windows | MSVC 2022 with vcpkg-managed `libpng`, `libjpeg-turbo`, `libtiff`, `libxml2`, `zlib`, `wxwidgets`, `nlohmann-json` |
+| Windows | MSVC 2022 or 2026 with vcpkg-managed `libpng`, `libjpeg-turbo`, `libtiff`, `libxml2`, `zlib`, `wxwidgets`, `nlohmann-json` |
 
 Thread support is provided by the platform C/C++ runtime and CMake's
 `Threads::Threads` imported target; no separate Ubuntu package is required.
@@ -58,6 +58,10 @@ cd iccdev
 cmake --preset macos-xcode -S Build/Cmake -B out/macos-xcode
 cmake --build out/macos-xcode --config Release -j"$(sysctl -n hw.ncpu)"
 ```
+
+Always provide `-S Build/Cmake` and an isolated `-B out/macos-xcode`
+directory. Configuring in the repository root is rejected to keep generated
+Xcode files out of the source tree.
 
 To open the generated project:
 
@@ -115,6 +119,18 @@ cmake --build out/vs2022-clangcl-x64 --config Release -- /m /maxcpucount
 
 The ClangCL preset uses the same `out\vs2022-clangcl-x64\bin\Release` runtime
 layout as the MSVC preset.
+
+## Windows MSVC 2026
+
+The `vs2026-x64` preset targets the Visual Studio 18 2026 generator and v145
+toolset while preserving the VS 2022 preset's vcpkg and runtime layout:
+
+```cmd
+git clone https://github.com/InternationalColorConsortium/iccDEV.git iccdev
+cd iccdev
+cmake --preset vs2026-x64 -S Build/Cmake -B out/vs2026-x64
+cmake --build out/vs2026-x64 --config Release -- /m /maxcpucount
+```
 
 ## Windows MinGW UCRT64
 

